File: two-sum/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype List[int]
        """
        d = {}
        if len(nums) < 1:
            return false
        for i in range(len(nums)):
            if nums[i] in d:
                return [d[nums[i]], i]
            d[target-nums[i]] = i

File: two-sum/two_sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    for k, v := range nums {
        if _, ok := m[target-v]; ok {
            if m[target-v] != k {
                return []int{m[target-v], k}
            }
        } else {
            m[v] = k
        }
    }
    return []int{}
}
